chore: remove usage of flask

dvmartinweigl opened the pull request:
Closes #248 

### Developer Checklist (Definition of Done)

**Issue**

- [ ] All acceptance criteria from the issue are met
- [ ] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [ ] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [ ] Branch is up-to-date with the branch to be merged with, i.e., develop
- [ ] Code is cleaned up and formatted
- [ ] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [ ] Descriptive title for this pull request is provided (will be used for release notes later)
- [ ] Reviewer and assignees are defined
- [ ] Add type label (e.g., *bug*, *feature*) to this pull request
- [ ] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [ ] The PR is connected to the corresponding issue (via `Closes #...`)
- [ ] [Summary of changes](#summary-of-changes) is written


### Summary of changes
- migrated xlsx api from flask to fastAPI
- migrated idtype_api from flask to fastAPI
- added missing backend tests
- removed unused endpoints from `mainapp.py`

### Screenshots


### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

thinkh reviewed (commented):
@dvmartinweigl and I have tested this PR with Ordino and it works after adding the removed werkzeug to Ordino's requirements.txt:

```
# pin werkzeug version because the new major version 3.0.0 breaks our applications on Oct 2 2023; the version can be removed later again when the other packages support v3.0.0
werkzeug==2.3.7
```

For the mock_data_api/requirements.txt we bumped the following dependencies:

```
uvicorn[standard]==0.29.0
fastapi==0.110.1
```

We also tested the up- and download for XLSX files and it works without problems.

dvmartinweigl commented:
I also tested this in bioinisght, sightline and aelixir and didn't encounter any issues.
Also covers testing for #247

puehringer commented on visyn_core/server/visyn_server.py:
```diff
@@ -139,12 +139,16 @@ def filter(self, record: logging.LogRecord) -> bool:
+            init_legacy_app(sub_app)
+            app.mount(namespace, WSGIMiddleware(sub_app))
+        except ImportError:
+            _log.error("Flask is not installed. Add flask to your project requirements if using legacy apps.")
```
@dvmartinweigl I think we can remove the error handling here, as I want to the app to fail if they use namespace plugins without installing Flask.

dvmartinweigl replied:
makes sense, removed it :+1:

puehringer commented on visyn_core/server/mainapp.py:
```diff
@@ -139,17 +33,6 @@ def build_info():
 
 
 # health check for docker-compose, kubernetes
+@mainapp_router.api_route("/health", methods=["GET", "HEAD"], include_in_schema=False)
```
@dvmartinweigl any reason why include_in_schema is false?

puehringer commented:
@dvmartinweigl regarding the /health being not available as it is not proxied. @thinkh and I just decided that we should not have any non-/api routes anymore. That means we have to refactor all of these to have a `/api` prefix. 

- [ ] All login mechanisms (i.e. they are proxied https://github.com/datavisyn/visyn_scripts/blob/main/config/rspack.config.js#L126-L137)
- [ ] /health (this needs a major release as it is currently being used in our helm charts)
- [ ] and anything I just missed. 

Feel free to do this in this PR.

dvmartinweigl commented:
> @dvmartinweigl regarding the /health being not available as it is not proxied. @thinkh and I just decided that we should not have any non-/api routes anymore. That means we have to refactor all of these to have a `/api` prefix.
> 
> * [x]  All login mechanisms (i.e. they are proxied https://github.com/datavisyn/visyn_scripts/blob/main/config/rspack.config.js#L126-L137)
> * [x]  /health (this needs a major release as it is currently being used in our helm charts)
> * [ ]  and anything I just missed.
> 
> Feel free to do this in this PR.

I just ensured all the routes to have the `/api` prefix and refactored the login mechanism.
I also found the `/metrics` endpoint to not have the `/api` prefix and added it, but wasn't sure if this affects our applications in any way??

I created a PR in visyn_scripts to remove the obsolete proxy for the security routes.
https://github.com/datavisyn/visyn_scripts/pull/67
